common/src/main/java/com/sequenceiq/cloudbreak/validation/CrnValidator.java

```diff
@@ -0,0 +1,55 @@
+package com.sequenceiq.cloudbreak.validation;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.google.common.base.Joiner;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnResourceDescriptor;
+
+public class CrnValidator implements ConstraintValidator<ValidCrn, String> {
```

Comment:
I feel like there is a lot of code duplication here with `CrnCollectionValidator`. Could you move the common parts into a common class?

Reply:
yes, makes sense